gptel.el

```diff
@@ -2829,7 +2951,13 @@ the response is inserted into the current buffer after point."
                              (kill-buffer buf)))
                          nil t nil)))
       ;; TODO: Add transformer here.
-      (setf (alist-get proc-buf gptel--request-alist) fsm))))
+      (setf (alist-get proc-buf gptel--request-alist)
+            (cons fsm
```

Comment:
We don't need to capture `proc-buf` into the closure, since this information is found in `gptel-abort` anyway and we can pass it as an argument to the abort-callback.  This should make the callbacks "static" functions that can be native-compiled.